src/decn/decn.c

```diff
@@ -1218,14 +1233,22 @@ inline void exp10_decn(void){
 	exp_decn();
 }
 
-//inline void pow_decn(void)
-//{
-//	copy_decn(&Tmp4Decn, &BDecn); //save b
-//	ln_decn();
-//	copy_decn(&BDecn, &Tmp4Decn); //restore b
-//	mult_decn(); //accum = b*ln(accum)
-//	exp_decn();
-//}
+inline void pow_decn(void) {
+	if (decn_is_zero(&BDecn)) {
+		copy_decn(&AccDecn, &DECN_1);
+		return;
+	}
+	if (decn_is_zero(&AccDecn)) {
+		set_dec80_zero(&AccDecn);
+		return;
+	}
+	//calculate AccDecn = AccDecn ^ BDecn
+	copy_decn(&Tmp4Decn, &BDecn); //save b
+	ln_decn();
+	copy_decn(&BDecn, &Tmp4Decn); //restore b
+	mult_decn(); //accum = b*ln(accum)
+	exp_decn();
+}
 
 
```

Comment:
The original reason I changed this from being `inline pow_decn()` in decn.c was it wasn't getting inlined properly and would cause the stack to overflow. This may not be an issue anymore. Can you confirm if you've tested on the actual calculator? I think the worst case is when calculating square roots, which are currently calculated using powers.

Reply:
Yes, I have it running on the actual hardware.  You are using `pow_decn` twice (also for sqrt). Hence, I thought non-inlining might be useful.  However, I have no idea how much pressure the stack is currently receiving.

Reply:
I've tested the latest commit without `pow_decn()` inlined, and it seems to be working alright now.